app/src/main/java/com/kotlin/readyplayerme/WebViewInterface.kt

```diff
@@ -7,20 +7,35 @@ import android.content.Context
 import android.content.Intent
 import android.webkit.JavascriptInterface
 import android.widget.Toast
+import org.json.JSONObject
 
 class WebViewInterface(private val context: Context) {
 
     @JavascriptInterface
-    fun openDialog(text: String){
-        // add to clipboard
-        val data = ClipData.newPlainText("Ready Player Me", text)
+    fun receiveData(text: String){
```

Comment:
Kotlin parses anything we send to it into `type string`?

Reply:
Maybe Kotlin has some auto casting ability, I noticed this now you mentioned, object sent, string received with to type errors. Weird.